src/arch/xtensa/include/arch/offset-defs.h

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright (c) 2017, Intel Corporation
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *   * Redistributions of source code must retain the above copyright
+ *     notice, this list of conditions and the following disclaimer.
+ *   * Redistributions in binary form must reproduce the above copyright
+ *     notice, this list of conditions and the following disclaimer in the
+ *     documentation and/or other materials provided with the distribution.
+ *   * Neither the name of the Intel Corporation nor the
+ *     names of its contributors may be used to endorse or promote products
+ *     derived from this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ */
+
+#include <xtensa/config/core-isa.h>
+
+#define OFFSET_EXCCAUSE     0x0
```

Comment:
These look like ABI offsets ? They should probably be in include/uapi/xtensa.h

Reply:
oh and the macros should be prefixed too (to make the names less generic) e.g. OFFSET_INTERRUPT

Reply:
@lgirdwood 
I don't think that userspace needs to access these offsets directly, as uapi/ipc/header.h defines sof_ipc_dsp_oops_xtensa message which should handle deserialization automagically, with any need of malually marching through offsets.
Also, could You clarify what kind of prefix would You like to have? Currently these names should reflect special registers' names, and that's what we've gone with.

Reply:
@akloniex ok, so these offsets are only for internal code ? not for any dump tool or script ?

Reply:
REG_OFFSET_ is a better descriptor prefix as it makes it obvious it's for registers.

Reply:
@lgirdwood sure, done